### PR TITLE
(case 20274) Create the HFM Preparation Step

### PR DESCRIPTION
**Pull request by sabrina-shanman**
https://highfidelity.manuscript.com/f/cases/20273/Improve-Model-Loading-Toolchain-Project-Phase-2
Sub-cases:
https://highfidelity.manuscript.com/f/cases/20274/Create-the-HFM-Preparation-Step

In Phase 1 of the Model Improvement Project, the format-specific loading of models was improved. As the start of Phase 2, this PR creates the HFM Preparation Step (aka Hifi Model Preparation Step) using the existing Task/Job/Engine system. This will become the new home for any model processing which is format-independent.

This PR is preceded by the last part of Phase 1, https://github.com/highfidelity/hifi/pull/14479, which added uniform MIME Type (Media Type) detection.

## TEST PLAN
- Verify models still load. This PR only changes model loading for interface in a very specific spot which affects all models equally, so if something bad happens it should be obvious.

### CODE REVIEW

File: libraries/model-networking/src/model-networking/ModelCache.cpp [hunk 1]
```diff
@@ -277,8 +278,12 @@ void GeometryDefinitionResource::downloadFinished(const QByteArray& data) {
...
 
 void GeometryDefinitionResource::setGeometryDefinition(HFMModel::Pointer hfmModel) {
-    // Assume ownership of the HFMModel pointer
-    _hfmModel = hfmModel;
+    // Do processing on the model
+    baker::Baker modelBaker(hfmModel);
+    modelBaker.run();
+
+    // Assume ownership of the processed HFMModel
+    _hfmModel = modelBaker.hfmModel;
```
Comment (sabrina-shanman) [inline libraries/model-networking/src/model-networking/ModelCache.cpp]: The model preparation step has been added to ModelCache.cpp here.

File: libraries/task/src/task/Task.h [hunk 2]
```diff
@@ -411,6 +411,7 @@ class Engine : public Task<JC, TP> {
     using Task = task::Task<ContextType, TimeProfiler>; \
     using Engine = task::Engine<ContextType, TimeProfiler>; \
     using Varying = task::Varying; \
+    template < typename T0 > using VaryingSet1 = task::VaryingSet1<T0>; \
```
Comment (sabrina-shanman) [inline libraries/task/src/task/Task.h]: I also added VaryingSet1 to the task namespace for consistency.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15874/HighFidelity-Beta-PR14564-566d09d.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15874/HighFidelity-Beta-Interface-PR14564-566d09d.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15874/HighFidelity-Beta-PR14564-566d09d.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15874/HighFidelity-Beta-Interface-PR14564-566d09d.exe

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15876/HighFidelity-Beta-PR14564-9e63caa.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15876/HighFidelity-Beta-Interface-PR14564-9e63caa.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15922/HighFidelity-Beta-PR14564-eaea398.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15924/HighFidelity-Beta-PR14564-c41174a.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15924/HighFidelity-Beta-PR14564-c41174a.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15924/HighFidelity-Beta-Interface-PR14564-c41174a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15924/HighFidelity-Beta-PR14564-c41174a.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15924/HighFidelity-Beta-Interface-PR14564-c41174a.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15949/HighFidelity-Beta-PR14564-a36e756.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15949/HighFidelity-Beta-PR14564-a36e756.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15949/HighFidelity-Beta-Interface-PR14564-a36e756.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15949/HighFidelity-Beta-PR14564-a36e756.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15949/HighFidelity-Beta-Interface-PR14564-a36e756.exe

**Comment by qaAdamSmith82** [comment]
Build this please

**Comment by qaAdamSmith82** [comment]
Currently Testing

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15987/HighFidelity-Beta-PR14564-a36e756.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15987/HighFidelity-Beta-PR14564-a36e756.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15987/HighFidelity-Beta-Interface-PR14564-a36e756.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15987/HighFidelity-Beta-PR14564-a36e756.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15987/HighFidelity-Beta-Interface-PR14564-a36e756.exe

**Comment by qaAdamSmith82** [comment]
Pre-Merge Test Passes

**Pull request merged**